src/server/game/Handlers/CharacterHandler.cpp

```diff
@@ -2541,6 +2541,12 @@ void WorldSession::HandleCharFactionOrRaceChangeCallback(std::shared_ptr<Charact
         }
     }
 
+    // Re-check all achievement criterias
```

Comment:
if (recvData.GetOpcode() == CMSG_CHAR_FACTION_CHANGE)
better？

Reply:
No. On race change should be the same.